src/lib/hooks/use-best-quote/utils/flashmint.ts

```diff
@@ -24,44 +23,62 @@ async function getEnhancedFlashMintQuote(
   isMinting: boolean,
   inputToken: Token,
   outputToken: Token,
-  indexTokenAmount: BigNumber,
+  indexTokenAmount: bigint,
   inputTokenPrice: number,
   outputTokenPrice: number,
   nativeTokenPrice: number,
   gasPrice: bigint,
   slippage: number,
   chainId: number,
   provider: IndexRpcProvider,
-  rpcUrl: string,
 ): Promise<Quote | null> {
   const inputTokenAddress = getAddressForToken(inputToken, chainId)
   const outputTokenAddress = getAddressForToken(outputToken, chainId)
+
   if (!inputTokenAddress || !outputTokenAddress) {
     console.warn('Error unkown input/output token')
     return null
   }
+
   const indexToken = isMinting ? outputToken : inputToken
   const inputOutputToken = isMinting ? inputToken : outputToken
+
   const currencies = getCurrencyTokensForIndex(indexToken, chainId)
   // Allow only supported currencies
   const isAllowedCurrency =
     currencies.filter((curr) => curr.symbol === inputOutputToken.symbol)
       .length > 0
-  if (!isAllowedCurrency) return null
+  if (!isAllowedCurrency) {
+    console.warn('Currency not allowed')
```

Comment:
nit: could log `inputOutputToken.symbol` in this message to make (possible) debugging easier